daemon/src/payout_curve/csr_tools.rs

```diff
@@ -149,4 +148,61 @@ mod tests {
 
         assert!(matches!(a, Error::CannotInitCSR));
     }
+
+    // test that all is good
+    #[test]
+    fn test_lu_matrix_solve_00() {
+        let a = CSR::new(
+            Array1::<f64>::from_vec(vec![11., 12., 22., 23., 31., 33.]),
+            Array1::<usize>::from_vec(vec![0, 1, 1, 2, 0, 2]),
+            Array1::<usize>::from_vec(vec![0, 2, 4, 6]),
+            (3, 3),
+        )
+        .unwrap();
+        let b = Array2::<f64>::from(vec![
+            [36., 59., 82.],
+            [204., 249., 294.],
+            [198., 262., 326.],
+        ]);
```

Comment:
This could be slightly more concise using the macro: https://docs.rs/ndarray/0.15.3/ndarray/macro.array.html